Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -669,6 +670,12 @@ ppExplanation (UnknownArch unknownArches) =
     "Unknown architecture name " ++ commaSep (map quote unknownArches)
 ppExplanation (UnknownCompiler unknownImpls) =
     "Unknown compiler name " ++ commaSep (map quote unknownImpls)
+ppExplanation (MissingUpperBounds names) =
+    "These packages miss upper bounds '"
+      ++ (intercalate "','" (unPackageName <$> names)) ++ "'"
+      ++ " please add them with with `cabal gen-bounds`."
+      ++ " For more information see: "
+      ++ " https://www.parsonsmatt.org/2020/05/07/on_pvp_restrictive_bounds.html"
```

Comment:
is parsonmatt better than — say — https://pvp.haskell.org/ ?

Reply:
I changed it to pvp, I originally did this post because it explains well why upperbounds are required, the pvp page only says you must do this, no reasoning, which is unsatisfying to me.

Reply:
the pvp faq has reasoning on upper bounds https://pvp.haskell.org/faq/#upper-bounds